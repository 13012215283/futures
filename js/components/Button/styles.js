import { StyleSheet } from 'react-native';

import { alpha } from '../../utils/color';
import { px2dp } from '../../utils/px2dp';
import { colors, sizes } from './../themes';

/**
 * 按钮根据规范有三种type：primary，operation及status
 * primary按钮：分为enable 和 disable
 * operation按钮：分为订单(order) 和 补款(fillMoney)
 * status按钮：未成交unsetted,交割中pending，已交割resolved，已撤单revoke及填入writeIn
 */
const primary = StyleSheet.create({
  container: {
    marginTop: px2dp(64),
    marginLeft: px2dp(64),
    marginRight: px2dp(64),
    borderRadius: px2dp(8),
    alignItems: 'center',
    justifyContent: 'center',
    height: px2dp(96),
  },
  text: {
    fontSize: sizes.f3,
    color: colors.white,
    textAlign: 'center',
  },
  enable: {
    backgroundColor: colors[1001],
  },
  disable: {
    backgroundColor: colors[1103],
  },
});

const operation = StyleSheet.create({
  container: {
    minWidth: px2dp(124),
    height: px2dp(60),
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: px2dp(4),
    borderWidth: StyleSheet.hairlineWidth,
  },
  text: {
    paddingLeft: px2dp(30),
    paddingRight: px2dp(30),
    fontSize: sizes.f2,
  },
  fillMoney: {
    borderColor: colors[1001],
    backgroundColor: alpha(colors[1001], 0.8),
  },
  order: {
    borderColor: colors[1103],
    backgroundColor: colors[1105],
  },
  fillMoneyText: {
    color: colors.white,
  },
  orderText: {
    color: colors[1102],
  },
});

const status = StyleSheet.create({
  container: {
    height: px2dp(44),
    borderRadius: px2dp(22),
    justifyContent: 'center',
    alignItems: 'center',
    borderStyle: 'solid',
    borderWidth: StyleSheet.hairlineWidth,
  },
  text: {
    textAlign: 'center',
    paddingLeft: px2dp(20),
    paddingRight: px2dp(20),
    alignItems: 'center',
    fontSize: sizes.f2,
    textAlignVertical: 'center',
    includeFontPadding: false,
  },
  unsettled: {
    borderColor: colors[1001],
    backgroundColor: alpha(colors[1001], 0.2),
  },
  unsettledText: {
    color: colors[1001],
  },
  pending: {
    borderColor: colors[1002],
    backgroundColor: alpha(colors[1002], 0.2),
  },
  pendingText: {
    color: colors[1002],
  },
  resolved: {
    borderColor: colors[1003],
    backgroundColor: alpha(colors[1003], 0.2),
  },
  resolvedText: {
    color: colors[1003],
  },
  revoke: {
    borderColor: colors[1103],
    backgroundColor: alpha(colors[1103], 0.2),
  },
  revokeText: {
    color: colors[1103],
  },
  writeIn: {
    height: px2dp(40),
    borderRadius: px2dp(20),
    borderColor: colors[1004],
    paddingLeft: px2dp(12),
    paddingRight: px2dp(12),
  },
  writeInText: {
    fontSize: sizes.f1,
    color: colors[1004],
  },
});

export { primary, status, operation };
