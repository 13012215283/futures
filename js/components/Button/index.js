import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity, Text, View, ViewPropTypes } from 'react-native';
import * as styles from './styles';

class Button extends Component {
  constructor(props) {
    super(props);
    this.onPress = this.onPress.bind(this);
  }

  onPress() {
    return this.props.onPress || null;
  }

  render() {
    const {
      type,
      text,
      subStatus,
      disabled,
      containerStyle,
      textStyle,
    } = this.props;
    const style = styles[type];
    let tStyle = [style.text];
    let cStyle = [style.container, style[subStatus]];

    // 按钮分为三种状态，对应容器样式
    if (type === 'operation' || type === 'status') {
      tStyle = [style.text, style[`${subStatus}Text`]];
    }

    tStyle = tStyle.concat([textStyle]);
    cStyle = cStyle.concat([containerStyle]);
    return (
      <TouchableOpacity disabled={disabled} onPress={this.props.onPress}>
        <View style={[cStyle, { overflow: 'hidden' }]}>
          <Text style={tStyle}>{text}</Text>
        </View>
      </TouchableOpacity>
    );
  }
}

Button.propTypes = {
  type: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  subStatus: PropTypes.string.isRequired,
  onPress: PropTypes.func,
  disabled: PropTypes.bool,
  containerStyle: ViewPropTypes.style,
  textStyle: Text.propTypes.style,
};

Button.defaultProps = {
  onPress() {},
  disabled: false,
  containerStyle: {},
  textStyle: {},
};

export default Button;
