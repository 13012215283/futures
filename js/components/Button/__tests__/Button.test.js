import React from 'react';
import TestRender from 'react-test-renderer';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import Button from '../../Button/';

test('should render correctly', () => {
  const button = TestRender.create(
    <View>
      <Button type="primary" text="确认平仓" subStatus="enable" />
      <Button
        type="status"
        text="交割中"
        subStatus="pending"
        containerStyle={{ height: 80 }}
        textStyle={{ color: '#ccc' }}
      />
    </View>
  ).toJSON();

  expect(button).toMatchSnapshot();
});

describe('<Button />', () => {
  it('simulates click events', () => {
    let click = false;
    const wrapper = shallow(
      <Button
        type="operation"
        text="平仓"
        subStatus="fillMoney"
        onPress={() => {
          click = true;
        }}
      />
    );
    wrapper.find('TouchableOpacity').simulate('press');
    expect(click).toBe(true);
  });
});
