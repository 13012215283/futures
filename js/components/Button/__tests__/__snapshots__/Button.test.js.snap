// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 8,
              "height": 96,
              "justifyContent": "center",
              "marginLeft": 64,
              "marginRight": 64,
              "marginTop": 64,
            },
            Object {
              "backgroundColor": "#e83b3b",
            },
            Object {},
          ],
          Object {
            "overflow": "hidden",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 32,
              "textAlign": "center",
            },
            Object {},
          ]
        }
      >
        确认平仓
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 22,
              "borderStyle": "solid",
              "borderWidth": 0.5,
              "height": 44,
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#30BE4133",
              "borderColor": "#30be41",
            },
            Object {
              "height": 80,
            },
          ],
          Object {
            "overflow": "hidden",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "alignItems": "center",
              "fontSize": 28,
              "includeFontPadding": false,
              "paddingLeft": 20,
              "paddingRight": 20,
              "textAlign": "center",
              "textAlignVertical": "center",
            },
            Object {
              "color": "#30be41",
            },
            Object {
              "color": "#ccc",
            },
          ]
        }
      >
        交割中
      </Text>
    </View>
  </View>
</View>
`;
